src/modules/job-info/list.c

```diff
@@ -135,26 +138,32 @@ json_t *list_jobs (struct info_ctx *ctx, int max_entries, json_t *attrs)
     /* We return jobs in the following order, pending, running,
      * inactive */
 
-    if ((ret = list_job_array (jobs,
-                               ctx->jsctx->pending,
-                               max_entries,
-                               attrs)) < 0)
-        goto error;
-
-    if (!ret) {
+    if (flags & FLUX_JOB_LIST_PENDING) {
         if ((ret = list_job_array (jobs,
-                                   ctx->jsctx->running,
+                                   ctx->jsctx->pending,
                                    max_entries,
                                    attrs)) < 0)
             goto error;
     }
 
-    if (!ret) {
-        if ((ret = list_job_array (jobs,
-                                   ctx->jsctx->inactive,
-                                   max_entries,
-                                   attrs)) < 0)
-            goto error;
+    if (flags & FLUX_JOB_LIST_RUNNING) {
+        if (!ret) {
+            if ((ret = list_job_array (jobs,
+                                       ctx->jsctx->running,
+                                       max_entries,
```

Comment:
oh good catch, that's a huge oversight by me.  Should add some unit tests too.

Reply:
Oh wait, things work b/c we are only returning only one jobs array to the user.  So the check below in `list_job_array()` does the trick.

```
            if (json_array_size (jobs) == max_entries)                                                                                                                                                             
                return 1;  
```

Reply:
OK, any way to make that behavior more evident in the function name or args?  (e.g. call it `list_job_append()` and pass `&jobs` with it initially set to NULL?   Since it's working no big deal if you leave as is.

Reply:
renaming to `list_job_append` seems like a good idea